Packs/TeamViewer/Integrations/TeamViewerEventCollector/TeamViewerEventCollector.py

```diff
@@ -0,0 +1,213 @@
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        client.http_request()
+        message = 'ok'
+    except DemistoException as e:
+        if 'Wrong' in str(e) or 'Authorization' in str(e):
+            message = 'Authorization Error: make sure bearer token format is correctly set'
+        else:
+            raise e
+    return message
+
+
+def fetch_events_command(
+    client: Client, max_fetch: int, last_run: Dict[str, Any], first_fetch_time: datetime
+) -> Tuple[Dict[str, Any], List[Dict[str, Any]]]:
+    """
+    Args:
+        client (Client): TeamViewer client to use.
+        max_fetch (int): The maximum number of events to fetch per log type.
+        last_run (dict): A dict with a keys containing the first event id to fetch for each log type.
+        first_fetch_time (str): In case of first fetch, fetch events from this date.
+    Returns:
+        dict: Next run dictionary containing the ids of the next events to fetch.
+        list: List of events that will be created in XSIAM.
+    """
+    # In the first fetch, get the ids for the first fetch time
+    demisto.debug('fetch_events_command start')
+    last_fetch = last_run.get('last_fetch')
+    last_fetch = first_fetch_time if last_fetch is None else datetime.strptime(last_fetch, DATE_FORMAT)
+    body = {
+        'StartDate': (last_fetch + timedelta(milliseconds=1)).strftime(DATE_FORMAT),
+        'EndDate': datetime.utcnow().strftime(DATE_FORMAT)
+    }
+    events, _ = search_events(client=client, limit=max_fetch, body=body)
+    next_run = {'last_fetch': max(events, key=lambda x: x['Timestamp'])['Timestamp']}
+    demisto.debug('fetch_events_command end')
+    return next_run, events
+
+
+def add_time_key_to_events(events: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    """
+    Adds the _time key to the events.
+    Args:
+        events: list, the events to add the time key to.
+    Returns:
+        list: The events with the _time key.
+    """
+    for event in events:
+        if event.get('Timestamp'):
+            event['_time'] = event.get('Timestamp')
+    return events
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    """
+    main function, parses params and runs command functions
+    """
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
+    api_key = params.get('credentials', {}).get('password')
+    base_url = urljoin(params.get('url'), '/api/v1/EventLogging')
+    verify_certificate = not params.get('insecure', True)
+    proxy = params.get('proxy', False)
+    demisto.debug('start')
+    # How much time before the first fetch to retrieve events
+    first_fetch_time: datetime = arg_to_datetime(
+        arg=params.get('first_fetch', '3 days'),
+        arg_name='First fetch time',
+        required=True,
+    )  # type: ignore   # datetime.datetime(2022, 1, 1, 00, 00, 00, 0)
```

Comment:
Do we need the datetime comment?

Reply:
not really, I thought it would be nice to have an example, removed